keyboard/_nixcommon.py

```diff
@@ -101,20 +101,44 @@ def __init__(self, devices, output=None):
         self.event_queue = Queue()
         self.devices = devices
         self.output = output or self.devices[0]
-        def start_reading(device):
-            while True:
-                self.event_queue.put(device.read_event())
-        for device in self.devices:
-            thread = Thread(target=start_reading, args=[device])
-            thread.daemon = True
-            thread.start()
 
+        for device in self.devices:
+            self.create_device_thread(device)
+        thread = Thread(target=self.check_for_new_devices)
+        thread.daemon = True
+        thread.start()
+
+    def check_for_new_devices(self):
+        while True:
+            devices_paths = [device.path for device in self.devices]
```

Comment:
please include time.sleep(2) so that this thread doesn't take up 100% of a single thread.

Reply:
Done, thanks for the remark!